rct229/rulesets/ashrae9012019/section19/section19rule10.py

```diff
@@ -0,0 +1,271 @@
+                hvac_id_b in baseline_system_types_dict_b[system_type]
+                for system_type in baseline_system_types_dict_b.keys()
+            )
+
+        def get_calc_vals(self, context, data=None):
+            hvac_b = context.baseline
+            hvac_id_b = hvac_b["id"]
+
+            HVAC_systems_primarily_serving_comp_rooms_list = data[
+                "HVAC_systems_primarily_serving_comp_rooms_list"
+            ]
+            baseline_system_types_b = list(data["baseline_system_types_dict_b"].keys())[
+                0
+            ]
+
+            fan_sys_b = hvac_b["fan_system"]
+            fan_air_economizer_b = find_one("$.air_economizer", fan_sys_b)
+
+            fan_air_economizer_type_b = find_one("$.air_economizer.type", fan_sys_b)
+
+            return {
+                "hvac_id_b": hvac_id_b,
+                "baseline_system_types_b": baseline_system_types_b,
+                "fan_air_economizer_b": fan_air_economizer_b,
+                "fan_air_economizer_type_b": fan_air_economizer_type_b,
+                "HVAC_systems_primarily_serving_comp_rooms_list": HVAC_systems_primarily_serving_comp_rooms_list,
+            }
+
+        def manual_check_required(self, context, calc_vals=None, data=None):
+            hvac_id_b = calc_vals["hvac_id_b"]
+            fan_air_economizer_b = calc_vals["fan_air_economizer_b"]
+            fan_air_economizer_type_b = calc_vals["fan_air_economizer_type_b"]
+            hvac_system_exception_2_list = data["hvac_system_exception_2_list"]
+
+            return (
+                (
+                    fan_air_economizer_b is None
+                    or fan_air_economizer_type_b == AIR_ECONOMIZER.FIXED_FRACTION
+                )
+                and hvac_id_b in hvac_system_exception_2_list
+            ) or (
+                fan_air_economizer_b is not None
+                and fan_air_economizer_type_b != AIR_ECONOMIZER.FIXED_FRACTION
+                and hvac_id_b in hvac_system_exception_2_list
+            )
+
+        def get_manual_check_required_msg(self, context, calc_vals=None, data=None):
+            hvac_id_b = calc_vals["hvac_id_b"]
+            baseline_system_types_b = calc_vals["baseline_system_types_b"]
+            fan_air_economizer_b = calc_vals["fan_air_economizer_b"]
+            fan_air_economizer_type_b = calc_vals["fan_air_economizer_type_b"]
+            hvac_system_exception_2_list = data["hvac_system_exception_2_list"]
+
+            if (
+                fan_air_economizer_b is None
+                or fan_air_economizer_type_b == AIR_ECONOMIZER.FIXED_FRACTION
+            ) and baseline_system_types_b in hvac_system_exception_2_list:
+                # Case 2 msg
+                undetermined_msg = f"Fail unless any of the zones served by the baseline system {hvac_id_b} in the proposed design include supermarket open refrigerated case-work systems that will be affected by using outdoor air for cooling (G3.1.2.6 exception #2)."
```

Comment:
Not sure why the message says Fail but it is actually Undetermined outcome?

Reply:
Good point. Addressed.